api/api_test.go

```diff
@@ -1,21 +1,24 @@
-package api
+package api_test
```

Comment:
```suggestion
package api
```
It's not a normal convention to have the package name as `api_test`. It should follow the convention as mentioned in https://blog.golang.org/package-names (short, clear, lower case, with no under_scores or mixedCaps).

Reply:
Actually, it's OK to have a test package of `xxx_test` indeed the standard go library itself uses such an idiom (eg. the `io_test` package inside the `io` package - https://github.com/golang/go/blob/master/src/io/io_test.go ). This helps with black box testing as it ensures that only exported functions and variables are tested, not internal ones. It also resolves issues with circular dependencies when using mocks.